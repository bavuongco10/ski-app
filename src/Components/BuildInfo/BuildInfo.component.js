import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import { buildText } from '../../Utils/buildInfo.util';
import { isProdEnv } from '../../Utils/environment.util';

const Text = styled.span`
  color: gray;
  font-size: 0.8em;
`;

const BuildInfo = ({ isProdEnv, buildText }) => {
  // we use 'production' to mean many things, including simply
  // local server vs. built artifacts. For this, we want to enable
  // render of build information except when *environment* is prod
  // if we're in prod, don't render:
  if (isProdEnv) return null;

  return <Text>{buildText}</Text>;
};

BuildInfo.propTypes = {
  isProdEnv: PropTypes.bool,
  buildText: PropTypes.string,
};

BuildInfo.defaultProps = {
  isProdEnv: isProdEnv(),
  buildText: buildText(),
};

export default BuildInfo;
