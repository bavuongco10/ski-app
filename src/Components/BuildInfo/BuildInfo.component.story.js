import React from 'react';
import { storiesOf } from '@storybook/react';

import BuildInfo from './BuildInfo.component';

storiesOf('BuildInfo', module)
  .add('test env', () => (
    <BuildInfo
      isProdEnv={false}
      buildText="sHa: vkjd392f | By: Bui Thanh Ba Vuong"
    />
  ))
  .add('production env', () => (
    <BuildInfo isProdEnv buildText="sHa: vkjd392f | By: Bui Thanh Ba Vuong" />
  ));
