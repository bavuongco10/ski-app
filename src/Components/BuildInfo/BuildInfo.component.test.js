import React from 'react';

import BuildInfo from './BuildInfo.component';

it('renders without crashing', () => {
  const wrapper = shallow(<BuildInfo />);
  expect(wrapper).toMatchSnapshot();
});
