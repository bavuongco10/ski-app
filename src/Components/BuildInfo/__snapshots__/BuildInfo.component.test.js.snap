// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<styled.span>
  SHA: undefined | By: undefined
</styled.span>
`;
